src/category_theory/limits/shapes/pullbacks.lean

```diff
@@ -2139,6 +2139,90 @@ by rw [← iso.eq_comp_inv, category.assoc, inr_inr_pushout_assoc_inv]
 
 end pushout_assoc
 
+section opposite
+
+open opposite
+
+/-- The pullback of `f` and `g` in `C` is isomorphic to the pushout of
+`f.op` and `g.op` in `Cᵒᵖ`. -/
+noncomputable
+def pullback_iso_unop_pushout (f : X ⟶ Z) (g : Y ⟶ Z)
+  [has_pullback f g] [has_pushout f.op g.op] : pullback f g ≅ unop (pushout f.op g.op) :=
+{ hom := (pushout.desc pullback.fst.op pullback.snd.op
+    (congr_arg quiver.hom.op pullback.condition) : pushout f.op g.op ⟶ op _).unop,
+  inv := pullback.lift (pushout.inl : _ ⟶ pushout f.op g.op).unop
+    (pushout.inr : _ ⟶ pushout f.op g.op).unop (congr_arg quiver.hom.unop
+      (@pushout.condition _ _ _ _ _ f.op g.op _)),
+  hom_inv_id' := by ext; simp [← unop_comp],
+  inv_hom_id' := by { apply quiver.hom.op_inj, ext; simp [← op_comp] } }
```

Comment:
I suggest you move this to the bottom of `limits/opposites.lean`. The proof is then just
```suggestion
def pullback_iso_unop_pushout' {X Y Z : C} (f : X ⟶ Z) (g : Y ⟶ Z)
  [has_pullback f g] [has_pushout f.op g.op] : pullback f g ≅ unop (pushout f.op g.op) :=
is_limit.cone_point_unique_up_to_iso (limit.is_limit _)
  ((pushout_cocone.is_colimit_equiv_is_limit_unop _) (colimit.is_colimit (span f.op g.op)))
```

Reply:
Done. It makes the lemmas that follow a bit harder to prove though, and I'm not sure if it is worth it.

Reply:
I think a big problem is that we are missing a bunch of API about pullback cones. I expect that if you add all variations of the following two lemmas, then `simp` should have a much easier time.
```lean
@[simp]
lemma pullback_cone.π_app_left {X Y Z : C} (f : X ⟶ Z) (g : Y ⟶ Z) (c : pullback_cone f g) :
  c.π.app walking_cospan.left = c.fst :=
rfl

@[simp]
lemma pushout_cocone.inl_colimit_cocone {X Y Z : C} (f : Z ⟶ X) (g : Z ⟶ Y)
  [has_colimit (span f g)]: pushout_cocone.inl (colimit.cocone (span f g)) = pushout.inl :=
rfl
```

Reply:
The main problem seems to be that the new definition introduces several identity maps and identity natural transforms here and there, and (presumably because of dependent type issues) simp fails to solve the otherwise trivial goal. The `dsimp, simp, dsimp, simp` trick solves some, but some needs `erw` (or `change`?) to work.

Reply:
My point is that all of these issues are already dealt with in the proofs of `pullback_cone.op_inr` etc., so if you can get those lemmas to fire, everything should work out file. The lemmas I mentioned above should hopefully enable `simp` to get the goal in a state where the lemmas apply.

Reply:
I see. Let me try.